### PR TITLE
Correctly import Caffe BatchNorm

### PR DESCRIPTION
**Pull request by matteosal**
The third blob of BatchNorm in Caffe was mistakenly interpreted as the moving average's exponential smoothing factor, while it's actually a scaling factor.
At test time, accumulated mean and average are divided by this factor before running the forward, as discussed [here](https://github.com/BVLC/caffe/issues/4885) ([code](https://github.com/BVLC/caffe/blob/master/src/caffe/layers/batch_norm_layer.cpp#L100-L105)). This PR rescales mean and variance during conversion to reproduce the intended forward behaviour.

Moreover, the variance regularization epsilon is extracted. Turns out that CuDNN requires that to be bigger than `CUDNN_BN_MIN_EPSILON`, currently set to `1e-5`. Is there a way to get it at conversion runtime?

@sbodenstein

### CODE REVIEW

**Comment by sbodenstein** [comment]
@matteo: we could do the following: if epsilon is smaller than or equal to `10^-5`, then set eps to `10^-4` and subtract the difference from the moving variance. This will guarantee that the predictions are the same as caffe, and that it can run on cuDNN.

**Comment by matteosal** [comment]
@sbodenstein: good suggestion, I added that. Numerical predictions are now fully restored (up to a usual ~10^-6 relative difference), altough the way it works is a bit ugly.
